### PR TITLE
feat(safes): add safes

### PR DESCRIPTION
**Pull request by qtotuan**

### CODE REVIEW

File: lib/v0/safes/base.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  state: {
+    ...oneOf({
+      type: 'object'
+      // properties to follow...
+    })
+  },
+  limit_upper: {
+    ...oneOf({
+      type: 'number',
+      description: 'Maximum amount that can be booked into the safe'
```
Comment (Endebert) [inline lib/v0/safes/base.js]: as these limits are money, might be a good idea to limit precision on those, i.e. `multipleOf: 0.01`

File: lib/v0/safes/base.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          currency: {
+            type: 'string',
+            description: 'The three letter [ISO currency](https://en.wikipedia.org/wiki/ISO_4217) of this item.',
+            example: '',
+            minLength: 3,
+            maxLength: 3
+          },
+          amount: {
+            type: 'number',
+            minimum: 0
```
Comment (Endebert) [inline lib/v0/safes/base.js]: same here regarding decimal precision.

File: lib/v0/safes/base.js (hunk 1 above)
Comment (qtotuan) [reply lib/v0/safes/base.js]: Makes sense - it's done

File: lib/v0/safes/base.js (hunk 2 above)
Comment (qtotuan) [reply lib/v0/safes/base.js]: Makes sense - it's done

**Pull request merged**